src/super_gradients/common/sg_loggers/dagshub_sg_logger.py

```diff
@@ -0,0 +1,222 @@
+        :param dagshub_auth:            Provide dagshub authentication token for automatic pipeline
+        :param log_mlflow_only:         Skip logging to DVC, use MLflow for all artifacts being logged
+        """
+        if monitor_system is not None:
+            logger.warning("monitor_system not available on DagsHubSGLogger. To remove this warning, please don't set monitor_system in your logger parameters")
+
+        self.s3_location_available = storage_location.startswith("s3")
+        super().__init__(
+            project_name=project_name,
+            experiment_name=experiment_name,
+            storage_location=storage_location,
+            resumed=resumed,
+            training_params=training_params,
+            checkpoints_dir_path=checkpoints_dir_path,
+            tb_files_user_prompt=tb_files_user_prompt,
+            launch_tensorboard=launch_tensorboard,
+            tensorboard_port=tensorboard_port,
+            save_checkpoints_remote=self.s3_location_available,
+            save_tensorboard_remote=self.s3_location_available,
+            save_logs_remote=self.s3_location_available,
+            monitor_system=False,
+        )
+        if _import_dagshub_error:
+            raise _import_dagshub_error
+
+        if _import_mlflow_error:
+            raise _import_mlflow_error
+        
+        self.repo_name, self.repo_owner, self.remote = None, None, None
+        if dagshub_repository:
+            self.repo_name, self.repo_owner = self.splitter(dagshub_repository)
+        
+        if dagshub_auth:
+          dagshub.auth.add_app_token(dagshub_auth)
+
+        self._init_env_dependency()
+
+        self.log_mlflow_only = log_mlflow_only
+        self.save_checkpoints_dagshub = save_checkpoints_remote
+        self.save_logs_dagshub = save_logs_remote
+
+    @staticmethod
+    def splitter(repo):
+        splitted = repo.split("/")
+        if len(splitted) != 2:
+            raise ValueError(f"Invalid input, should be owner_name/repo_name, but got {repo} instead")
+        return splitted[1], splitted[0]
+
+    def _init_env_dependency(self):
+        """
+        The function creates paths for the DVC directory, models, and artifacts, obtains an authentication token from Dagshub, and sets MLflow tracking credentials. 
+        It also checks whether the repository name and owner have been set and prompts the user to enter them if they haven't. 
+        If the remote URI is not set or does not include "dagshub", Dagshub is initialized with the repository name and owner, and the remote URI is obtained. 
+        The method then creates a Repo object with the repository information and sets the DVC folder to the DVC directory path. 
+        """
+
+        self.paths = {
+            "dvc_directory": Path("artifacts"),
```

Comment:
Where will these directories be created? 
Please make sure they are created inside the `checkpoints_dir_path`

Reply:
This is actually one of the cool benefits of using DagsHub with DVC. These are all directories that are created on the DagsHub repo, without touching the local environment, so this can run from anywhere without creating files locally.

Reply:
ok. thanks